lib/datasource/docker/index.ts

```diff
@@ -535,3 +540,22 @@ export async function getPkgReleases({
   }
   return ret;
 }
+
+function getECRAuthToken(region: string, opts: hostRules.HostRule) {
+  const config = { region, accessKeyId: undefined, secretAccessKey: undefined };
+  if (opts.username && opts.password) {
+    config.accessKeyId = opts.username;
+    config.secretAccessKey = opts.password;
+  }
+  const ecr = new AWS.ECR(config);
+  return new Promise<string>((resolve, reject) => {
+    ecr.getAuthorizationToken({}, (err, data) => {
+      if (err) {
+        logger.error({ err });
+        reject(err);
```

Comment:
How is our authorization header going to look in this situation?

Reply:
ie what happens if `getAuthorizationToken` fails? I'm guessing it would fail to determine the tags for this image and skip. Needs a test case (this PR in general does - and I'm guessing thats why it's not passing the checks due to coverage threshold not being met)

Reply:
So in the case where we attempt to get an authorization token and fail to do so, what do we want to do? Should we try without the header?

Reply:
It certainly can. I've made that change now and there's a test case demonstrating it